src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,261 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useConnect, useConnectors } from 'wagmi';
+import { closeSvg } from '../../internal/svg/closeSvg';
+import { coinbaseWalletSvg } from '../../internal/svg/coinbaseWalletSvg';
+import { defaultAvatarSVG } from '../../internal/svg/defaultAvatarSVG';
+import { walletConnectSvg } from '../../internal/svg/walletConnectSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  line,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useOnchainKit } from '../../useOnchainKit';
+
+type WalletModalProps = {
+  isOpen: boolean; // Controls the visibility state of the modal
+  onClose: () => void; // Callback function to close the modal
+  className?: string; // Optional className override for the element
+};
+
+export function WalletModal({ isOpen, onClose, className }: WalletModalProps) {
+  // We need to keep the component mounted during the exit animation
+  const [shouldRender, setShouldRender] = useState(false);
+  const { connect } = useConnect();
+  const connectors = useConnectors();
+  const { config } = useOnchainKit();
+
+  useEffect(() => {
+    if (isOpen) {
+      // Immediately render the modal when opening
+      setShouldRender(true);
+    } else {
+      // When closing, wait for animation to complete before removing from DOM
+      const timer = setTimeout(() => {
+        setShouldRender(false);
+      }, 100);
```

Comment:
I think you should be able to use a transitionEnd event here instead.

Reply:
good call